src/libAtomVM/opcodesswitch.h

```diff
@@ -4970,6 +5046,105 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
+
+                    int size = 2 + n_freeze;
+                    if (memory_ensure_free(ctx, size + 1) != MEMORY_GC_OK) {
+                        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+                    }
+                    term *boxed_func = memory_heap_alloc(ctx, size + 1);
+
+                    boxed_func[0] = (size << 6) | TERM_BOXED_FUN;
+                    boxed_func[1] = (term) mod;
+                    boxed_func[2] = term_from_int(fun_index);
+                #endif
+
+                int size_args;
+                DECODE_INTEGER(size_args, code, i, next_off, next_off);
+                for (int j = 0; j < size_args; j++) {
+                    term arg;
+                    DECODE_COMPACT_TERM(arg, code, i, next_off, next_off);
+                    #ifdef IMPL_EXECUTE_LOOP
+                        boxed_func[3 + j] = arg;
+                    #endif
+                }
+
+                #ifdef IMPL_EXECUTE_LOOP
+                    term fun = ((term) boxed_func) | TERM_BOXED_VALUE_TAG;
+                    WRITE_REGISTER(dreg_type, dreg, fun);
+                #endif
+                NEXT_INSTRUCTION(next_off);
+                break;
+            }
+
+            case OP_INIT_YREGS: {
+                int next_off = 1;
+                next_off++; // skip extended list tag
+                int size;
+                DECODE_INTEGER(size, code, i, next_off, next_off);
+                for (int j = 0; j < size; j++) {
+                    int target;
+                    DECODE_INTEGER(target, code, i, next_off, next_off);
+                    #ifdef IMPL_EXECUTE_LOOP
+                        ctx->e[target] = term_nil();
+                    #endif
+                }
+                NEXT_INSTRUCTION(next_off);
+                break;
+            }
+
+            case OP_RECV_MARKER_BIND: {
+                int next_off = 1;
+                dreg_t reg_a;
+                dreg_type_t reg_a_type;
+                DECODE_DEST_REGISTER(reg_a, reg_a_type, code, i, next_off, next_off);
+                dreg_t reg_b;
+                dreg_type_t reg_b_type;
+                DECODE_DEST_REGISTER(reg_b, reg_b_type, code, i, next_off, next_off);
+                NEXT_INSTRUCTION(next_off);
+                break;
+            }
+
+            case OP_RECV_MARKER_CLEAR: {
+                int next_off = 1;
+                dreg_t reg_a;
+                dreg_type_t reg_a_type;
+                DECODE_DEST_REGISTER(reg_a, reg_a_type, code, i, next_off, next_off);
+                NEXT_INSTRUCTION(next_off);
+                break;
+            }
+
+            case OP_RECV_MARKER_RESERVE: {
+                int next_off = 1;
+                dreg_t reg_a;
+                dreg_type_t reg_a_type;
+                DECODE_DEST_REGISTER(reg_a, reg_a_type, code, i, next_off, next_off);
+                NEXT_INSTRUCTION(next_off);
+                break;
+            }
+
+            case OP_RECV_MARKER_USE: {
```

Comment:
Did you run a coverage tool to assess these new opcodes are covered by existing tests?

Reply:
yes

![coverage](https://user-images.githubusercontent.com/273180/203633934-415d6d36-7b20-4851-a448-50a2790a846a.png)

Reply:
the last version ([734d4b4](https://github.com/atomvm/AtomVM/pull/429/commits/734d4b4e81e03500ddb0d1903dfafc54c0dce457)):

1. new instructions are covered
2. there are only two occurrences of DECODE_ALLOCATOR_LIST now - for both occurrences tests cover the case of having extended lists (verified manually)